edit/linter-meta.js

```diff
@@ -0,0 +1,51 @@
+/* global linter */
+'use strict';
+
+function createMetaCompiler(cm) {
+  const successCallbacks = [];
+  let meta = null;
+  let metaIndex = null;
+  let cache = [];
+
+  linter.register((text, options, _cm) => {
+    if (_cm !== cm) {
+      return;
+    }
+    const match = text.match(/\/\*\s*==userstyle==[\s\S]*?==\/userstyle==\s*\*\//i);
+    if (!match) {
+      return [];
+    }
+    if (match[0] === meta && match.index === metaIndex) {
+      return cache;
+    }
+    return parseMeta(match[0])
+      .then(result => {
+        for (const cb of successCallbacks) {
+          cb(result);
+        }
+        meta = match[0];
+        metaIndex = match.index;
+        cache = [];
+        return cache;
+      }, err => {
+        meta = match[0];
+        metaIndex = match.index;
+        cache = [{
+          from: cm.posFromIndex((err.index || 0) + match.index),
+          to: cm.posFromIndex((err.index || 0) + match.index),
+          message: err.message,
+          severity: 'error'
+        }];
+        return cache;
+      });
+  });
+
+  return {
+    onSuccess: cb => successCallbacks.push(cb)
```

Comment:
Some modules emit different types of events e.g. `linter` emits `change` and `unhook` events. For consistency, it might be a good idea to keep the method as `'on' + eventName`.

Or maybe we can adopt an event library e.g. [event-lite](https://www.npmjs.com/package/event-lite)?

Reply:
onSuccess is weird because it implies there should be onFailure. AFAICT there'll be no need for any other events so we can just use a single listener. Otherwise it should be renamed to something like onUpdated. As for an event library, I doubt we need it for such a trivial use case.